main_test.go

```diff
@@ -209,7 +209,61 @@ func TestTransferFungibleTokensToTreasuryActions(t *testing.T) {
 }
 
 func TestTransferNonFungibleTokensToAccountActions(t *testing.T) {
-	// TODO
+	var transferNFTActionUUID uint64
+
+	otu := NewOverflowTest(t)
+	//create NFT collection and mint NFT for signer1
+	otu.CreateNFTCollection("signer1")
+	otu.MintNFT("signer1")
+
+	//set up treasury and send nft from signer1 to treasuray
+	otu.SetupTreasury("treasuryOwner", Signers, uint64(DefaultThreshold))
+	otu.CreateNFTCollection("treasuryOwner")
+	otu.SendCollectionToTreasury("treasuryOwner", "treasuryOwner")
+	otu.SendNFTToTreasury("signer1", "treasuryOwner", 0)
+
+	//set up the account
+	otu.CreateNFTCollection("account");
+
+
+	t.Run("Signers should be able to propose a transfer of non fungible tokens out of the Treasury", func(t *testing.T) {
+		otu.ProposeNonFungibleTokenTransferAction("treasuryOwner", Signers[0], "account", uint64(0))
+	})
+
+	t.Run("Signers should be able to sign to approve a proposed action to transfer a non-fungible token to an account", func(t *testing.T) {
+		// Get first ID of proposed action
+		actions := otu.GetProposedActions("treasuryOwner")
+		keys := make([]uint64, 0, len(actions))
+		for k := range actions {
+			keys = append(keys, k)
+		}
+		transferNFTActionUUID = keys[0]
+
+		// Each signer submits an approval signature
+		for _, signer := range Signers {
+			otu.SignerApproveAction("treasuryOwner", transferNFTActionUUID, signer)
+		}
+
+		// Assert that the signatures were registered
+		signersMap := otu.GetVerifiedSignersForAction("treasuryOwner", transferNFTActionUUID)
+		for _, signer := range Signers {
+			assert.True(otu.T, true, signersMap[otu.GetAccountAddress(signer)])
+		}
+	})
+
+	t.Run(`A signer should be able to execute a proposed action to transfer a non-fungible token to an account once it has received
+		the required threshold of signatures`, func(t *testing.T) {
+		otu.ExecuteAction("treasuryOwner", transferNFTActionUUID)
+
+		// Assert that the NFT has been transfered out of the treasury vault
+		collectionIds := otu.GetTreasuryIdentifiers("treasuryOwner")
+		ownedNFTIds := otu.GetTreasuryCollection("treasuryOwner", collectionIds[1][0])
+		assert.Equal(otu.T, 0, len(ownedNFTIds))
+
+		// TODO: Assert that the NFT has been transfered into the account collection
```

Comment:
Try implementing this TODO!

You'll need to write your own cadence script.

Reply:
👌 I'm going to 
1. Create a `GetAccountNFTCollection` function in `test_utils.go` and a `get_account_collection.cdc` script, and have the function call the script. Is this the approach we want to take?

Reply:
You got it!